### PR TITLE
add: add numeric ID generation

### LINKED ISSUE #1: Add numeric ID generation

### PR DESCRIPTION
**Pull request by PhyTensor**
@thismaker check it out

the branch can be deleted thereafter

### CODE REVIEW

File: src/Cryptography/CryptoUtility.cs
Comment (PhyTensor) [inline src/Cryptography/CryptoUtility.cs:1]: I'm assuming that the line I added is appropriate - it looks appropriate

**Review by PhyTensor — COMMENTED** [review]
@thismaker  check this too

**Comment by thismaker** [comment]
There are a few problems:
1. The signature of the function is `Random(Int)` but there is already another method with a similar signature i.e `Random(Int,bool)`. The second method has an optional bool, so therefore calling `string s = CryptoUtility.Random(6)` would cause some ambiguity as to which overload is being referred to.
2. The documentation is not clear, "a random numeric string" sounds better than "a random string of numbers"
3. There was no need to directly call `_random.Next`, there is already a function `Random(int min, int max)` included in the class, I'd prefer you call that function unless you have a better reason not to?

File: src/Cryptography/CryptoUtility.cs [hunk 1]
```diff
@@ -115,6 +117,22 @@ public static string Random(int length, bool lowerCase = false)
...
+        /// <returns>A randomly generated string of numerics</returns>
+        public static string Random(int length)
+        {
+            // Min = 10^(n - 1)
+            // Max = 10^(n) - 1
+            // Where n = length
+            int min = Convert.ToInt32( Math.Pow( 10, ( length - 1 ) ) );
+            int max = Convert.ToInt32( Math.Pow( 10, length ) - 1 );
+
+            return _random.Next(min, max).ToString(CultureInfo.InvariantCulture);
```
Comment (thismaker) [inline src/Cryptography/CryptoUtility.cs]: you can use the preexisting function  instead, see comment for details

File: src/Cryptography/CryptoUtility.cs [hunk 2]
```diff
@@ -115,6 +117,22 @@ public static string Random(int length, bool lowerCase = false)
 
         }
 
+        /// <summary>
+        /// Generates a random string of numbers of specified <paramref name="length"/>
```
Comment (thismaker) [inline src/Cryptography/CryptoUtility.cs]: Prefer: random numeric string

File: src/Cryptography/CryptoUtility.cs [hunk 3]
```diff
@@ -115,6 +117,22 @@ public static string Random(int length, bool lowerCase = false)
...
+        /// </summary>
+        /// <param name="length">The number of numerics to include in the string</param>
+        /// <returns>A randomly generated string of numerics</returns>
+        public static string Random(int length)
```
Comment (thismaker) [inline src/Cryptography/CryptoUtility.cs]: Name might cause ambiguity

**Review by thismaker — CHANGES_REQUESTED** [review]

**Comment by PhyTensor** [comment]
> There are a few problems:
> 
> 1. The signature of the function is `Random(Int)` but there is already another method with a similar signature i.e `Random(Int,bool)`. The second method has an optional bool, so therefore calling `string s = CryptoUtility.Random(6)` would cause some ambiguity as to which overload is being referred to.
> 2. The documentation is not clear, "a random numeric string" sounds better than "a random string of numbers"
> 3. There was no need to directly call `_random.Next`, there is already a function `Random(int min, int max)` included in the class, I'd prefer you call that function unless you have a better reason not to?

working on them

**Review by PhyTensor — COMMENTED** [review]
@thismaker Corrections have been made

**Comment by thismaker** [comment]
great, now bump the package version all the csprojs then you can merge

**Review by PhyTensor — COMMENTED** [review]
version updated to 0.0.3

**Pull request merged**